src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -194,8 +194,9 @@ var ReactElementValidator = {
         false,
         'React.createElement: type should not be null, undefined, boolean, or ' +
           'number. It should be a string (for DOM elements) or a ReactClass ' +
-          '(for composite components).%s',
-        getDeclarationErrorAddendum()
+          '(for composite components).%s%s',
+        getDeclarationErrorAddendum(),
+        ReactComponentTreeHook.getCurrentStackAddendum()
```

Comment:
The format is

```
in Button (at DangerButton.js:123)
```

In this case, we would not have the component name because the source would say `<Foo />` but Foo is just an empty object `{}` or something else invalid. If the jsx source babel transform is used then we will have the file/line but otherwise we won't. Theoretically we could make another babel transform that also makes the variable name "Foo" introspectable but it would probably be useful for only this warning so it's hard to justify it.